### PR TITLE
refactor: use `@std/tar` in `build` task and Deno 2 in CI

### PR DESCRIPTION
**Pull request by iuioiua**
And updates CI to use Deno 2.

Towards https://github.com/denoland/std/issues/5986

### CODE REVIEW

**Comment by iuioiua** [comment]
Please re-open. This should be rebased once the next release of `std` is cut, and then merged.

File: lib/wasmopt.ts [hunk 1]
```diff
@@ -1,10 +1,7 @@
 // Copyright 2018-2024 the Deno authors. MIT license.
 
-import { Buffer } from "@std/io/buffer";
-import { Untar } from "@std/archive/untar";
-import { copy } from "@std/io/copy";
+import { UntarStream } from "https://raw.githubusercontent.com/denoland/std/0352a3c441aefe39d1233b4be92c49e404963457/tar/untar_stream.ts";
```
Comment (kt3k) [inline lib/wasmopt.ts:3]: Can this be `@std/tar/untar-stream` now?

Comment (iuioiua) [reply lib/wasmopt.ts]: Fixed now.

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**